src/main/kotlin/no/nav/familie/ba/mottak/integrasjoner/BaseService.kt

```diff
@@ -0,0 +1,72 @@
+package no.nav.familie.ba.mottak.integrasjoner
+
+import no.nav.familie.log.NavHttpHeaders
+import no.nav.familie.log.mdc.MDCConstants
+import no.nav.security.token.support.client.core.ClientProperties
+import no.nav.security.token.support.client.core.oauth2.OAuth2AccessTokenService
+import no.nav.security.token.support.client.spring.ClientConfigurationProperties
+import org.slf4j.MDC
+import org.springframework.boot.web.client.RestTemplateBuilder
+import org.springframework.http.*
+import org.springframework.http.client.ClientHttpRequestExecution
+import org.springframework.http.client.ClientHttpRequestInterceptor
+import org.springframework.http.client.ClientHttpResponse
+import org.springframework.util.LinkedMultiValueMap
+import org.springframework.util.MultiValueMap
+import org.springframework.web.client.RestOperations
+import org.springframework.web.client.exchange
+import java.net.URI
+import java.util.*
+
+class BearerAuthorizationInterceptor(private val oAuth2AccessTokenService: OAuth2AccessTokenService, private val clientProperties: ClientProperties) : ClientHttpRequestInterceptor {
+    override fun intercept(request: HttpRequest, body: ByteArray, execution: ClientHttpRequestExecution): ClientHttpResponse {
+        val response = oAuth2AccessTokenService.getAccessToken(clientProperties)
+
+        request.headers.setBearerAuth(response.accessToken)
+        return execution.execute(request, body)
+    }
+}
+
+open class BaseService(clientConfigKey: String, restTemplateBuilder: RestTemplateBuilder,
+                       clientConfigurationProperties: ClientConfigurationProperties,
+                       oAuth2AccessTokenService: OAuth2AccessTokenService) {
+    private val clientProperties: ClientProperties = Optional.ofNullable(
```

Comment:
Trenger ikke Optional. Kan skrives 
private val clientProperties: ClientProperties = clientConfigurationProperties.registration[clientConfigKey] ?: throw RuntimeException("could not find oauth2 client config for key=$clientConfigKey")